zerocopy-derive/src/lib.rs

```diff
@@ -250,15 +234,33 @@ pub fn derive_known_layout(ts: proc_macro::TokenStream) -> proc_macro::TokenStre
 pub fn derive_no_cell(ts: proc_macro::TokenStream) -> proc_macro::TokenStream {
     let ast = syn::parse_macro_input!(ts as DeriveInput);
     match &ast.data {
-        Data::Struct(strct) => {
-            impl_block(&ast, strct, Trait::NoCell, RequireBoundedFields::Yes, false, None, None)
-        }
-        Data::Enum(enm) => {
-            impl_block(&ast, enm, Trait::NoCell, RequireBoundedFields::Yes, false, None, None)
-        }
-        Data::Union(unn) => {
-            impl_block(&ast, unn, Trait::NoCell, RequireBoundedFields::Yes, false, None, None)
-        }
+        Data::Struct(strct) => impl_block(
+            &ast,
+            strct,
+            Trait::NoCell,
+            FieldBounds::All(&[Trait::NoCell]),
```

Comment:
`Trait::Self` would require adding panic paths in `impl_block` (namely, it would be invalid to pass `Trait::Self` as the `trt` argument - the trait to be implemented), so I added a separate `TraitBound` enum that should have the same effect without introducing panic paths. Lmk if that looks better.